### PR TITLE
[GUI] Support text_colored in GGUI

### PR DESCRIPTION
**Pull request by gaocegege**
Signed-off-by: Ce Gao <ce.gao@outlook.com>

Related issue = #3013 

- [x] Implement the feature
- [x] Add docs and comments
- [x] Add examples

Feature preview ./examples/ggui_examples/mass_spring_game_ggui.py

```diff
-        window.GUI.text(
+        window.GUI.text_colored((0,112,112,112),
            "Left click: add mass point (with shift to fix); Right click: attract"
        )
```

Screenshot from 2021-10-01 21-28-12

### CODE REVIEW

**Comment by gaocegege** [comment]
The PR is ready to review.

File: examples/ggui_examples/mass_spring_game_ggui.py [hunk 1]
```diff
@@ -26,6 +26,9 @@
 rest_length = ti.field(dtype=ti.f32,
                        shape=(max_num_particles, max_num_particles))
 
+# gray color
+gray = (128, 128, 128, 255)
```
Comment (AmesingFlank) [inline examples/ggui_examples/mass_spring_game_ggui.py:30]: All the other GGUI APIs use floats between 0 and 1 to represent color. The `text_colored` API should do the same for consistency.

**Review by AmesingFlank — APPROVED** [review]
LGTM! Many thanks for contributing :D

Please take a look at the color range issue between merging the PR.

By the way, GGUI have tests now. See `tests/python/test_ggui.py`. Although, the test depend on images stored in the repo, so we wouldn't want to update the tests very often and unnecessarily inflate the git history size. If you plan to add further IMGUI components, we can wait until all of them are finished and write tests for them together.

File: examples/ggui_examples/mass_spring_game_ggui.py (hunk 1 above)
Comment (AmesingFlank) [reply examples/ggui_examples/mass_spring_game_ggui.py:30]: There's a chance that IMGUI automatically guesses the color range used (see the existing `color_edit_3` API) so perhaps nothing needs to be done here. But please investigate and confirm this.

**Comment by gaocegege** [comment]
> By the way, GGUI have tests now. See tests/python/test_ggui.py. Although, the test depend on images stored in the repo, so we wouldn't want to update the tests very often and unnecessarily inflate the git history size. If you plan to add further IMGUI components, we can wait until all of them are finished and write tests for them together.

Sure, thanks for the information!

File: examples/ggui_examples/mass_spring_game_ggui.py (hunk 1 above)
Comment (gaocegege) [reply examples/ggui_examples/mass_spring_game_ggui.py:30]: OK I will double check it.

**Comment by gaocegege** [comment]
Sorry, I don't have the bandwidth to fix the PR, I am closing it

**Pull request closed without merging**